providers/sendgrid/src/lib/sendgrid.provider.ts

```diff
@@ -70,12 +74,22 @@ export class SendgridEmailProvider implements IEmailProvider {
   private createMailData(options: IEmailOptions) {
     const dynamicTemplateData = options.customData?.dynamicTemplateData;
     const templateId = options.customData?.templateId as unknown as string;
+    const asm = options.customData?.asm as ASMConfig;
+    if (asm?.groupId) {
+      asm.groupId = parseInt(asm.groupId as any); // Sendgrid expects groupId to be a number
+    }
+    if (asm?.groupsToDisplay) {
+      asm.groupsToDisplay = asm.groupsToDisplay.map((group) =>
+        parseInt(group as any)
+      ); // Sendgrid expects groupsToDisplay to be an array of numbers
+    }
```

Comment:
✏ **suggestion:** if we would like to allow passing the string or number for both fields, please create a mapper function that will create a new object with appropriate value types like: 

```
  private mapAsmOptions(asm: ASMConfig): MailDataRequired['asm'] {
    return {
      groupId: parseInt(`${asm.groupId}`),
      groupsToDisplay: asm.groupsToDisplay.map((group) => parseInt(`${group}`)),
    };
  }
...

const asm = this.mapAsmOptions(options.customData?.asm);
```